File: src/main/java/com/pancisin/bookster/models/Notification.java
package com.pancisin.bookster.models;

import java.util.Calendar;
import java.util.UUID;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import org.hibernate.annotations.GenericGenerator;

import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "notifications")
public class Notification {

	@Id
	@GeneratedValue(generator = "uuid2")
	@GenericGenerator(name = "uuid2", strategy = "uuid2")
	@Column(updatable = false, nullable = false, columnDefinition = "BINARY(16)")
	private UUID id;

	@JsonIgnore
	@ManyToOne
	private User recipient;

	@Column
	private String code;
	
	@Column
	private boolean seen = false;

	@Column(name = "created", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP", insertable = false, updatable = false)
	private Calendar created;

	@Column
	private String target;
	
	public Notification() {

	}

	public Notification(String code, String target) {
		this.code = code;
		this.target = target;
	}
	
	public Notification(String code) {
		this.code = code;
	}

	public UUID getId() {
		return id;
	}

	public User getRecipient() {
		return recipient;
	}

	public boolean isSeen() {
		return seen;
	}

	public void setSeen(boolean seen) {
		this.seen = seen;
	}

	public Calendar getCreated() {
		return created;
	}

	public String getTarget() {
		return target;
	}

	public String getCode() {
		return code;
	}
}

File: src/main/kotlin/com/pancisin/bookster/model/Notification.kt
package com.pancisin.bookster.model

import java.util.Calendar
import java.util.UUID

import org.hibernate.annotations.GenericGenerator

import com.fasterxml.jackson.annotation.JsonIgnore
import com.pancisin.bookster.model.enums.NotificationType
import javax.persistence.*

@Entity
@Table(name = "notifications")
data class Notification(
  @Id @GeneratedValue(generator = "uuid2") @GenericGenerator(name = "uuid2", strategy = "uuid2")
  @Column(updatable = false, nullable = false, columnDefinition = "BINARY(16)")
  val id: UUID? = null,

  @JsonIgnore @ManyToOne
  var recipient: User? = null,

  @Column
  var code: String? = null,

  @Column
  var seen: Boolean = false,

  @Column(name = "created", columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP", insertable = false, updatable = false)
  val created: Calendar? = null,

  @Column
  var target: String? = null,

  @Column
  var subject: String? = null,

  @Column
  @Enumerated(EnumType.STRING)
  var type: NotificationType = NotificationType.INFO
) {
  constructor(code: String, target: String, subject: String) : this(null, null, code, false, null, target, subject)
  constructor(code: String, target: String) : this(null, null, code, false, null, target)
  constructor(code: String) : this(null, null, code)
}
